dynamic-relations/src/main/java/at/drm/service/DynamicRelationsPrintService.java

```diff
@@ -0,0 +1,127 @@
+        stack.push(root);
+        StringBuilder result = new StringBuilder("\n");
+        var levelByNode = new HashMap<TreeNodeRelationIdentity, Integer>();
+        levelByNode.put(root, 0);
+        int level;
+        while (!stack.isEmpty()) {
+            var parentObject = stack.pop();
+            level = levelByNode.get(parentObject);
+            result.append(" ".repeat(level)).append(parentObject.object().getType()).append("\n");
+            for (var child : parentObject.childObjects()) {
+                stack.push(child);
+                levelByNode.put(child, level + 1);
+            }
+        }
+        log.info(result.toString());
+    }
+
+    private void createTree(TreeNodeRelationIdentity root) {
+        Set<Relation> visited = new HashSet<>();
+        Deque<TreeNodeRelationIdentity> stack = new ArrayDeque<>();
+        stack.push(root);
+        while (!stack.isEmpty()) {
+            var parentObject = stack.poll();
+            Collection<RelationLink> relations = tryGetSourceObjectRelations(parentObject);
+            TreeNodeRelationIdentity childObject;
+            for (RelationLink relation : relations) {
+                var entityClass = classBySimpleName.get(relation.getTargetType());
+                var childRelationIdentity = entityManager.find(entityClass, relation.getTargetId());
+                childObject = new TreeNodeRelationIdentity(childRelationIdentity, new LinkedList<>());
+                var r = new Relation(parentObject.object(), childObject.object());
+                if (!visited.contains(r)) {
+                    parentObject.childObjects().add(childObject);
+                    visited.add(r);
+                    stack.add(childObject);
+                }
+            }
+        }
+    }
+
+    @PostConstruct
+    public void configure() {
+        Set<Class<RelationIdentity>> classesImplementingInterface = findClassesImplementingInterface();
+        classBySimpleName = classesImplementingInterface.stream()
+            .collect(Collectors.toMap(clazz -> clazz.getSimpleName() + "Type", Function.identity()));
+    }
+
+
+    private <T> Set<Class<T>> findClassesImplementingInterface() {
+        Set<Class<T>> classes = new HashSet<>();
+
+        ClassPathScanningCandidateComponentProvider scanner =
+            new ClassPathScanningCandidateComponentProvider(false);
+
+        scanner.addIncludeFilter(new AssignableTypeFilter(RelationIdentity.class));
+        Set<BeanDefinition> candidates = scanner.findCandidateComponents("*");
+        for (BeanDefinition candidate : candidates) {
+            try {
+                Class<T> clazz = (Class<T>) Class.forName(candidate.getBeanClassName());
+                if (!clazz.isInterface() && !Modifier.isAbstract(clazz.getModifiers())) {
+                    classes.add(clazz);
+                }
+            } catch (ClassNotFoundException e) {
+                System.err.println("Class not found: " + candidate.getBeanClassName());
```

Comment:
Use the log.error instead of the System.err
I think it should also throw an Runtime Exception